include/common/map_coordinate.h

```diff
@@ -98,11 +97,89 @@ struct Cartesian
 {
     typedef D Datum;
 
-    const units::length::meter_t x;
-    const units::length::meter_t y;
-    const units::length::meter_t z;
+    units::length::meter_t x;
+    units::length::meter_t y;
+    units::length::meter_t z{ 0 };
+
+    auto operator+(const Cartesian<D> &coords)
+    {
+        Cartesian<D> out;
+        out.x = x + coords.x;
+        out.y = y + coords.y;
+        out.z = z + coords.z;
+        return out;
+    };
+
+    void operator+=(const Cartesian<D> &coords)
+    {
+        x = x + coords.x;
+        y = y + coords.y;
+        z = z + coords.z;
+        
+    };
+};
+
+//----------------------------------------------------------------------------
+// BoBRobotics::MapCoordinate::Ellipsoid
+//----------------------------------------------------------------------------
+//! A latitude and longitude relative to a particular datum
+template<typename D>
+struct LatLon
+{
+    typedef D Datum;
+
+    units::angle::degree_t lat;
+    units::angle::degree_t lon;
+    units::length::meter_t height{ 0 }; // Set default value, because we don't always want to use height
+
 };
 
+//! A standard GPS coordinate
+using GPSCoordinate = LatLon<WGS84>;
+
+//! Adds shift to a LatLon<Datum> by converting to Cartesian and adding
+//! a Cartesian v to it. 
+template<typename Datum>
+inline LatLon<Datum> shiftLatLon (LatLon<Datum> &latLon, Cartesian<Datum> &v)
+{
+    LatLon<Datum> target = cartesianToLatLon(latLonToCartesian(latLon) + v);
+    return target;
+}
+
+//! Adds shift to a UTM Coordinate by converting to LatLon<Datum> and then
+//! to Cartesian. Than add a Cartesian v to it and transfer back.
+//! a Cartesian v to it. 
+template<typename Datum>
+inline UTMCoordinate shiftUTM (UTMCoordinate &utm, Cartesian<Datum> &v)
+{
+    // Setup return value
+    UTMCoordinate target;
+    target.height = utm.height;
+
+    // Convert UTM to lat long 
+    double lat;
+    double lon;
+    using UTM::UTMtoLL;
+    UTMtoLL(utm.northing.value(),utm.easting.value(),
+            utm.zone,lat,lon);
+    
+    // Convert LL members to unit library classes
+    LatLon<Datum> G;
+    G.lat = degree_t(lat);
+    G.lon = degree_t(lon);
+    G.height = utm.height;
+
+    // Calculate new position in terms of lat long
+    G = shiftLatLon(G,v);
+    
+    // Convert back to UTM
+    using UTM::LLtoUTM;
```

Comment:
If you only use it once, I would say there's no point in seperately bringing it into your local namespace, just call it with ``UTM::LLtoUTM``